### PR TITLE
zephyr: prefix `autoconf.h` with `zephyr/`

### PR DESCRIPTION
**Pull request by ycsin**
Update the include path of `autoconf.h` to `zephyr/autoconf.h`

See https://github.com/zephyrproject-rtos/zephyr/pull/63973

### CODE REVIEW

**Comment by ycsin** [comment]
Ping @dleach02

File: s32/mcux/devices/S32K344/S32K344_features.h [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #ifndef _S32K344_FEATURES_H_
 #define _S32K344_FEATURES_H_
 
-#include "autoconf.h"
+#include <zephyr/autoconf.h>
```
Comment (decsny) [inline s32/mcux/devices/S32K344/S32K344_features.h:10]: Will this break CI until the zephyr side merges?

File: mcux/mcux-sdk/cmake/mcux.cmake [hunk 2]
```diff
@@ -26,7 +26,7 @@ set(APPLICATION_BINARY_DIR ${CMAKE_SOURCE_DIR}/build CACHE PATH
     "Application Binary Directory"
 )
 
-set(AUTOCONF_H ${APPLICATION_BINARY_DIR}/kconfig/include/generated/autoconf.h)
+set(AUTOCONF_H ${APPLICATION_BINARY_DIR}/kconfig/include/generated/zephyr/autoconf.h)
```
Comment (decsny) [inline mcux/mcux-sdk/cmake/mcux.cmake:29]: I believe this file is not actually being used by zephyr

File: mcux/mcux-sdk/scripts/kconfig/kconfiglib.py
Comment (decsny) [inline mcux/mcux-sdk/scripts/kconfig/kconfiglib.py:1]: I believe this file is also not actually being used by zephyr, we need to remove it

File: s32/mcux/devices/S32K344/S32K344_features.h (hunk 1 above)
Comment (ycsin) [reply s32/mcux/devices/S32K344/S32K344_features.h:10]: #63973 has been merged last week

**Review by decsny — APPROVED** [review]

File: s32/mcux/devices/S32K344/S32K344_features.h (hunk 1 above)
Comment (dleach02) [inline s32/mcux/devices/S32K344/S32K344_features.h:10]: @manuargue @Dat-NguyenDuy

This is in your area

**Review by Dat-NguyenDuy — APPROVED** [review]

**Pull request merged**